packages/common/src/server-connection.js

```diff
@@ -55,31 +74,36 @@ export const Provider = ({ Pusher, pusherKey, debug = false, children }) => {
     if (accessToken == null || user?.id == null) return;
     Pusher.logToConsole = debug;
 
-    const pusher = new Pusher(pusherKey, {
-      cluster: "eu",
-      authEndpoint: `${apiOrigin}/websockets/auth`,
-      auth: {
-        params: { provider: "pusher" },
-        headers: { Authorization: `Bearer ${accessToken}` },
-      },
-    });
-
-    setPusherState(pusher.connection.state);
+    const connect = async () => {
```

Comment:
Since we now wait until pusher establishes a connection until we set the first pusher state (with `setPusherState` at the bottom here) we won’t get any spinners when refreshing tokens. We just let the previous connection stay open until the new one is done connecting.